components/aws/actions/dynamodb-put-item/dynamodb-put-item.mjs

```diff
@@ -1,118 +1,99 @@
-        "SIZE",
-        "NONE",
-      ],
-    },
-    conditional_operator: {
-      type: "string",
-      description: "This is a legacy parameter. Use `ConditionExpression` instead. For more information, see [ConditionalOperator](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/LegacyConditionalParameters.ConditionalOperator.html) in the *Amazon DynamoDB Developer Guide*.",
-      optional: true,
-    },
-    condition_expression: {
-      type: "string",
-      description: "A condition that must be satisfied in order for a conditional `PutItem` to succeed. An expression can contain any of the following: functions, comparison operators, logical operators. For more information about condition expressions, see [Condition Expressions](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.SpecifyingConditions.html) in the *Amazon DynamoDB Developer Guide*.",
-      optional: true,
-    },
-    expression_attributeNames: {
-      type: "object",
-      description: "One or more substitution tokens for attribute names in an expression. The following are some use cases for using `ExpressionAttributeNames`:\n\n* To access an attribute whose name conflicts with a DynamoDB reserved word.\n* To create a placeholder for repeating occurrences of an attribute name in an expression.\n* To prevent special characters in an attribute name from being misinterpreted in an expression.\n\nUse the # character in an expression to dereference an attribute name.",
-      optional: true,
-    },
-    expression_attribute_values: {
-      type: "object",
-      description: "One or more values that can be substituted in an expression. Use the : (colon) character in an expression to dereference an attribute value. For more information on expression attribute values, see [Condition Expressions](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.SpecifyingConditions.html) in the *Amazon DynamoDB Developer Guide*.",
-      optional: true,
+  },
+  methods: {
+    async tableAttributeDefinitions(region, tableName) {
+      const response = await this.aws.dynamodbDescribeTable(region, {
+        TableName: tableName,
+      });
+      return response.Table.AttributeDefinitions;
     },
   },
-  async run() {
-    //See the API docs: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB.html#putItem-property
+  async additionalProps() {
+    const props = {};
+    if (this.tableName) {
+      const [
+        primaryKey,
+        secondaryKey,
+      ] = await this.tableAttributeDefinitions(this.region, this.tableName);
+      props.primaryKey = {
+        type: "string",
+        label: primaryKey.AttributeName,
+        description: "Value for the primary key",
+      };
+      if (secondaryKey) {
+        props.secondaryKey = {
+          type: "string",
+          label: secondaryKey.AttributeName,
+          description: "Value for the sort key",
+        };
+      }
+      props.item = {
```

Comment:
`Item` doesn't need to be an additional prop, but when put here it maintains a better prop order in the UI. A feature for ordering props would be great for these cases